src/libopenrave-core/generictrajectory.cpp

```diff
@@ -441,6 +441,17 @@ class GenericTrajectory : public TrajectoryBase
             WriteBinaryVector(O, this->_vtrajdata);
 
             WriteBinaryString(O, GetDescription());
+
+            if( !GetReadableInterfaces().empty() ) {
```

Comment:
@strmio this is not a good way to maintain a binary format by looking for any left over data and trying to match that to a string. You should instead:
1. update traj version and preserve back compatibility
2. write the number of readable interfaces, and then write the string contents of each as a binary blob

When reading, you should read the number of interfaces, and for every blob of data, you should create a `StringXMLReadable` interface for every readable interface.

Reply:
@strmio ping

Reply:
You mean it is okay to put aside XML formatted readable interfaces?